nucleus/deployment/common/src/main/java/org/glassfish/deployment/common/DeploymentException.java

```diff
@@ -67,7 +67,7 @@ public DeploymentException(Throwable t)
             // the DeploymentException travels between processes and needs
```

Comment:
I know that this comment suggests that they don't want to use the normal `super(Throwable)` mechanism. I would suggest that this is faulty logic. All exceptions should be serializable (as `Throwable` implements `Serializable`. Perhaps it could use this method, and then perform some logic to make sure every chained exception is serializable?

Reply:
Aha, they throw away all internal causes, create new exception and only use stacktrace of the last exception, so deployer would never see internals. But also does not know what caused the failure.
Nevermind, I would vote for standard mechanism. Usually the person deploying the application is admin user.
The problem might be some old RMI client and deserialization, but I don't know noone who would use it today. Maybe deployment-client, JSR-88? We can try it ;)